### ISSUE TITLE
[Feature] Create virtual files for compilations

### ISSUE THREAD

**Issue by HarleyQuinn-Z6** [stashapp/stash#2511]
If you have a scene that is a compilation and you have the source of the clips in the compilation it would be nice to substitute the compilation for a virtual file that had the links, time and duration to mark the start and end of the clip in each of the source scenes. That way instead of storing the compilation you would only have to store those links, times and durations. I don't know if this is possible so I can't give any more insight into how to accomplish it.

**Comment by JediMaster25** [comment]
Another possibility of virtual files would be to make random compilations, given a few tags it could use clips from different scenes that use those tags. For this, markers would need to have a beginning and an end, to be able to use them as a clip.

**Comment by xx790** [comment]
This!
I need this.

Use cases:

- Cock Hero/Fap Hero videos produced as continuous works but can also be enjoyed when favorite individual rounds are taken individually and remixed. Storing copies of all videos in loose rounds is very inefficient.
- Have access to favorite clips to put into a queue that won't necessarily be full scenes, and again, avoid manual editing and disk space waste.

I don't mind whether this will be an extension to markers as long as it won't put unexpected constraints.

**Comment by JediMaster25** [comment]
Also I think there should be a new `markers_tags` table in the database instead of a single `primary_tag_id` column in the `scenes_markers` table. To be able to select markers based on multiple tags.

**Comment by mf35** [comment]
There will be so much potential with the markers if you give them an end timestamp.

In VLC player there is a way to create generated compilations based on random samples of a video playlist.
With the addon "[Samples playlist generator](https://addons.videolan.org/p/1154039)"

Problems are :
- This is not accurate: you set fixed durations and begin / end timestamps, same for all the samples.
Or random timestamps.
- Playing all the samples is not seamless, VLC player can't preload videos (Potplayer can do that but i don't think such a samples generator exists with this player) so it reads the samples juming from a video to another, and to a position to another, and most of the time you have some "black frames" or "first frame of the video" between each samples.
In short: not a perfect cut-free play.

Anyway, if we can have this "marker video clips" feature in Stash, we could create playlist with them and then create powerful compilations with ease.

**Comment by laurus-lx** [comment]
I think this relates to a proposal for stash db schema revision that would support "File Ranges" with start/end times [198667](https://github.com/stashapp/stash/discussions/1958#discussioncomment-2549480).

File Ranges are sub-sections/clips, and scenes can be composed of many such file-ranges.
